comp/process/agent/status_test.go

```diff
@@ -0,0 +1,141 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package agent
+
+import (
+	"bytes"
+	"embed"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//go:embed fixtures
+var fixturesTemplates embed.FS
+
+func fakeStatusServer(t *testing.T, errCode int, response []byte) *httptest.Server {
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
+
+		if errCode != 200 {
+			http.NotFound(w, r)
+		} else {
+			_, err := w.Write(response)
+			require.NoError(t, err)
+		}
+	}
+
+	return httptest.NewServer(http.HandlerFunc(handler))
+}
+
+func TestStatus(t *testing.T) {
```

Comment:
Not sure if having this test is worthwhile. It's nearly the same as https://github.com/DataDog/datadog-agent/blob/main/comp/process/status/statusimpl/status_test.go. Might be worthwhile if the behavior diverges more in the future.